github/repos.go

```diff
@@ -1007,27 +1007,31 @@ func (s *RepositoriesService) ListAllTopics(ctx context.Context, owner, repo str
 		return nil, resp, err
 	}
 
-	return topics, resp, nil
+	return topics.Names, resp, nil
 }
 
 // ReplaceAllTopics replaces topics for a repository.
 //
 // GitHub API docs: https://developer.github.com/v3/repos/#replace-all-topics-for-a-repository
-func (s *RepositoriesService) ReplaceAllTopics(ctx context.Context, owner, repo string, topics *Topics) (*Topics, *Response, error) {
+func (s *RepositoriesService) ReplaceAllTopics(ctx context.Context, owner, repo string, topics []string) ([]string, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/topics", owner, repo)
-	req, err := s.client.NewRequest("PUT", u, topics)
+	t := new(Topics)
+	if t.Names = topics; topics == nil {
+		t.Names = []string{}
+	}
```

Comment:
I think the above 4 lines can be expressed in a more readable way:

```Go
t := &Topics{
	Names: topics,
}
if t.Names == nil {
	t.Names = []string{}
}
```

Reply:
Removed ternary operator.